diff --git a/hypr/hypridle.conf b/hypr/hypridle.conf
index 36f8b62b..712f9b09 100644
--- a/hypr/hypridle.conf
+++ b/hypr/hypridle.conf
@@ -11,16 +11,16 @@ general {
 
 listener {
     timeout = 300 # 5mins
-    on-timeout = loginctl lock-session
+    on-timeout = [ -f /tmp/qs_inhibit ] || loginctl lock-session
 }
 
 listener {
     timeout = 600 # 10mins
-    on-timeout = hyprctl dispatch 'hl.dsp.dpms({ action = "disable" })'
+    on-timeout = [ -f /tmp/qs_inhibit ] || hyprctl dispatch 'hl.dsp.dpms({ action = "disable" })'
     on-resume = hyprctl dispatch 'hl.dsp.dpms({ action = "enable" })'
 }
 
 listener {
     timeout = 900 # 15mins
-    on-timeout = $suspend_cmd
+    on-timeout = [ -f /tmp/qs_inhibit ] || $suspend_cmd
 }
