diff --git a/quickshell/ii/modules/ii/verticalBar/VerticalBarContent.qml b/quickshell/ii/modules/ii/verticalBar/VerticalBarContent.qml
index b9689be8..41982836 100644
--- a/quickshell/ii/modules/ii/verticalBar/VerticalBarContent.qml
+++ b/quickshell/ii/modules/ii/verticalBar/VerticalBarContent.qml
@@ -149,7 +149,27 @@ Item { // Bar content region
                 Layout.fillWidth: true
                 Layout.fillHeight: false
             }
-            
+
+            HorizontalBarSeparator {
+                visible: controllerBattery.available
+            }
+
+            ControllerBattery {
+                id: controllerBattery
+                Layout.fillWidth: true
+                Layout.fillHeight: false
+            }
+
+            HorizontalBarSeparator {
+                visible: budsBattery.available
+            }
+
+            BudsBattery {
+                id: budsBattery
+                Layout.fillWidth: true
+                Layout.fillHeight: false
+            }
+
         }
     }
 
